analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/SymbolicValue.cs

```diff
@@ -74,5 +69,59 @@ public sealed record SymbolicValue
             Constraints.Any()
                 ? Constraints.Values.Select(x => x.ToString()).OrderBy(x => x).JoinStr(", ")
                 : "No constraints";
+
+        private static SymbolicValue AddOrReplaceConstraint(SymbolicValue baseValue, SymbolicConstraint constraint)
+        {
+            if (baseValue.HasConstraint(constraint))
+            {
+                return baseValue;
+            }
+
+            var constraintCount = baseValue.Constraints.Count;
+            if (constraintCount == 0 || (constraintCount == 1 && baseValue.Constraints.ContainsKey(constraint.GetType())))
+            {
+                return SingleConstraint(constraint);
+            }
+
+            return baseValue with { Constraints = baseValue.Constraints.SetItem(constraint.GetType(), constraint) };
+        }
+
+        private static SymbolicValue RemoveConstraint(SymbolicValue baseValue, SymbolicConstraint constraint) =>
```

Comment:
The `static` causes more trouble then it solves. Let's remove it everywhere
```suggestion
        private SymbolicValue RemoveConstraint(SymbolicConstraint constraint) =>
```

Reply:
I inlined the overloads.